test/unit/attributes-test.ts

```diff
@@ -47,6 +47,14 @@ describe('Model attributes', function() {
     expect(author.relationships['books']).to.deep.eq([])
   });
 
+  it('has enumerable properties', function() {
+    let person = new Person();
+    let keys = Object.keys(person)
+
+    expect(keys).to.include('firstName');
+    expect(keys).to.include('lastName');
+  });
```

Comment:
I think this is the case, but verifying that `class BadPerson extends Person` can still enumerate the properties of the `Person` class?

Reply:
I'll add a test for that, good call.